Furball.Vixie.Backends.Direct3D11/Direct3D11Backend.cs

```diff
@@ -1,128 +1,174 @@
 #endif
 
-            Device device = new Device(DriverType.Hardware, deviceFlags, featureLevel);
-            DeviceContext deviceContext = device.ImmediateContext;
+            D3D11.D3D11CreateDevice(null, DriverType.Hardware, deviceFlags, new[] { featureLevel }, out this._device, out this._deviceContext);
 
-            this._device        = device;
-            this._deviceContext = deviceContext;
+#if DEBUG
+            this._device.QueryInterface<ID3D11Debug>().ReportLiveDeviceObjects(ReportLiveDeviceObjectFlags.Detail);
+#endif
 
-            SwapChainDescription1 swapChainDescription = new SwapChainDescription1 {
-                Width  = window.Size.X,
-                Height = window.Size.Y,
-                Format = Format.R8G8B8A8_UNorm,
+            IDXGIFactory dxgiFactory = this._device.QueryInterface<IDXGIDevice>().GetParent<IDXGIAdapter>().GetParent<IDXGIFactory>();
+
+            int i = 0;
+            try {
+                while (dxgiFactory.GetAdapter(i) != null) {
+                    AdapterDescription description = dxgiFactory.GetAdapter(i).Description;
+
+                    long luid = description.Luid.LowPart | description.Luid.HighPart;
+
+                    string dedicatedSysMemMb = Math.Round((description.DedicatedSystemMemory / 1024.0) / 1024.0, 2).ToString(CultureInfo.InvariantCulture);
+                    string dedicatedVidMemMb = Math.Round((description.DedicatedVideoMemory  / 1024.0) / 1024.0, 2).ToString(CultureInfo.InvariantCulture);
+                    string dedicatedShrMemMb = Math.Round((description.SharedSystemMemory    / 1024.0) / 1024.0, 2).ToString(CultureInfo.InvariantCulture);
+
+                    Logger.Log($"///////////////////Adapter [{i}]////////////////////", LoggerLevelD3D11.InstanceInfo);
+                    Logger.Log($"Adapter Description:       {description.Description}", LoggerLevelD3D11.InstanceInfo);
+                    Logger.Log($"Revision:                  {description.Revision}",    LoggerLevelD3D11.InstanceInfo);
+                    Logger.Log($"PCI Vendor ID:             {description.VendorId}",    LoggerLevelD3D11.InstanceInfo);
+                    Logger.Log($"PCI Device ID:             {description.DeviceId}",    LoggerLevelD3D11.InstanceInfo);
+                    Logger.Log($"PCI Subsystem ID:          {description.SubsystemId}", LoggerLevelD3D11.InstanceInfo);
+                    Logger.Log($"Locally Unique Identifier: {luid}",                    LoggerLevelD3D11.InstanceInfo);
+                    Logger.Log($"Dedicated System Memory:   {dedicatedSysMemMb}mb",     LoggerLevelD3D11.InstanceInfo);
+                    Logger.Log($"Dedicated Video Memory:    {dedicatedVidMemMb}mb",     LoggerLevelD3D11.InstanceInfo);
+                    Logger.Log($"Dedicated Shared Memory:   {dedicatedShrMemMb}mb",     LoggerLevelD3D11.InstanceInfo);
+                    Logger.Log($"//////////////////////////////////////////////////\n", LoggerLevelD3D11.InstanceInfo);
```

Comment:
We should really provide a standardized interface for printing info like this, GL, D3D, and Veldrid all now do it differently

Reply:
should that be part of this PR though? i agree with what ur saying but that i feel is out of scope for this pr

Reply:
> should that be part of this PR though? i agree with what ur saying but that i feel is out of scope for this pr

i guess it can stay out of this PR